kostyor/tests/unit/upgrades/test_engine.py

```diff
@@ -132,133 +128,102 @@ def test_multinode_assignment(self):
+            mock.call({'name': 'heat-api',
+                       'version': constants.MITAKA},
+                      [hosts[1]]),
+            mock.call({'name': 'heat-engine',
+                       'version': constants.MITAKA},
+                      [hosts[1]]),
+            mock.call({'name': 'heat-api-cfn',
+                       'version': constants.MITAKA},
+                      [hosts[1]]),
+            mock.call({'name': 'heat-api-cloudwatch',
+                       'version': constants.MITAKA},
+                      [hosts[1]]),
+
             # compute
-            mock.call(self.upgrade,
-                      {'name': 'nova-compute',
-                       'version': constants.MITAKA}),
-            mock.call(self.upgrade,
-                      {'name': 'neutron-linuxbridge-agent',
-                       'version': constants.MITAKA}),
+            mock.call({'name': 'nova-compute',
+                       'version': constants.MITAKA},
+                      [hosts[0]]),
+            mock.call({'name': 'neutron-linuxbridge-agent',
+                       'version': constants.MITAKA},
+                      [hosts[0]]),
 
             # storage
-            mock.call(self.upgrade,
-                      {'name': 'cinder-volume',
-                       'version': constants.MITAKA}),
-        ]
-        self.assertEqual(
-            expected_service_calls,
-            self.engine.driver.pre_service_upgrade_hook.call_args_list)
-        self.assertEqual(
-            expected_service_calls,
-            self.engine.driver.start_upgrade.call_args_list)
-        self.assertEqual(
-            expected_service_calls,
-            self.engine.driver.post_service_upgrade_hook.call_args_list)
-
-        expected_host_calls = [
-            mock.call(self.upgrade,
-                      {'hostname': 'host-2',
-                       'cluster_id': '2ba8fad8-3a0f-47db-a45b-62df1d811687',
-                       'id': '5708c51f-5421-4ecd-9a9e-000000000002'}),
-            mock.call(self.upgrade,
-                      {'hostname': 'host-1',
-                       'cluster_id': '2ba8fad8-3a0f-47db-a45b-62df1d811687',
-                       'id': '5708c51f-5421-4ecd-9a9e-000000000001'}),
-            mock.call(self.upgrade,
-                      {'hostname': 'host-3',
-                       'cluster_id': '2ba8fad8-3a0f-47db-a45b-62df1d811687',
-                       'id': '5708c51f-5421-4ecd-9a9e-000000000003'}),
+            mock.call({'name': 'cinder-volume',
+                       'version': constants.MITAKA},
+                      [hosts[2]]),
         ]
         self.assertEqual(
-            expected_host_calls,
-            self.engine.driver.pre_host_upgrade_hook.call_args_list)
-        self.assertEqual(
-            expected_host_calls,
-            self.engine.driver.post_host_upgrade_hook.call_args_list)
+            expected_start_calls,
+            self.engine.driver.start.call_args_list)
```

Comment:
Why not 'assert_has_calls'? https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_has_calls